spec/requests/api/v2/asset_audits_spec.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require './spec/requests/api/v2/shared_examples/api_key_authenticatable'
+
+describe 'AssetAudits API', with: :api_v2 do
+  let(:base_endpoint) { '/api/v2/asset_audits' }
+
+  it_behaves_like 'ApiKeyAuthenticatable'
+
+  context 'with multiple AssetAudits' do
+    before { create_list(:asset_audit, 5) }
+
+    it 'returns a list of asset audits' do
+      api_get base_endpoint
+
+      expect(response).to have_http_status(:success)
+      expect(json['data'].length).to eq(5)
+    end
+
+
+  end
+
+  context 'with an AssetAudit' do
+    let(:resource_model) { create(:asset_audit) }
+    let(:updated_resource_model) do
+      resource_model.update(witnessed_by: 'new_witness_user')
+      resource_model
+    end
+
+
+    let(:payload) do
+      {
+        'data' => {
+          'id' => resource_model.id,
+          'type' => 'asset_audits',
+          'attributes' => {
+            'witnessed_by' => updated_resource_model.witnessed_by,
+          }
+        }
+      }
+    end
+
+    it 'returns an individual AssetAudit' do
+      api_get "#{base_endpoint}/#{resource_model.id}"
+      expect(response).to have_http_status(:success)
+      expect(json.dig('data', 'type')).to eq('asset_audits')
+      expect(json.dig('data','attributes', 'key')).to eq(resource_model.key)
+      expect(json.dig('data', 'attributes','witnessed_by')).to eq(resource_model.witnessed_by)
+    end
+
+    it 'allows update of an AssetAudit' do
```

Comment:
I thought our implementation stopped updates?  Perhaps it doesn't work an the other model objects that implement this aren't working either.

Reply:
Hmmm, I wonder if that's because of the mapping from the `asset_uuid` to actual assets.  Looking at the definition for a plate resource, it has something about `default_include :barcode` which may be how it is able to filter the values?  Might be a red herring.  Sorry I don't really know, I'm just spit balling.

It looks in some cases you need to define how a filter should be applied: https://jsonapi-resources.com/v0.9/guide/resources.html#Applying-Filters

Perhaps that's needed?  Given that this endpoint was added specifically for Asset Audits and it is working there, I wonder whether this is a problem to leave for another time when we actually need the functionality.  Just make a note of it in a comment so someone else doesn't have to spend ages finding out it was hard to solve.